### PR TITLE
Exclude all .txt files, analyze by category, add luggage to sales_ranking

### PR DESCRIPTION
**Pull request by sulaimonlasisi**
This PR includes the following changes:
1. Instead of excluding specific .txt files, exclude all of them
2. Print out analyzed items by category. This helps us look specifically in categories we can sell. It also makes it easy to find an item because we know where to look.
3. Found a new category luggage and added it to the sales_ranking.js file
4. Some errors were coming up in AmazonProductList where product.Products.Product was undefined, so added a check for that.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,11 +1,5 @@
...
-# text files
-special_feeds_items.txt
-paginated_special_feeds_items.txt
-subcategory_info.txt
-category_info.txt
-amazon_items.txt
-paired_items.txt
-analyzed_items_info.txt
+# all text files
+*.txt
```
Comment (danfstucky) [inline .gitignore:5]: Ha, this was a good idea

File: src/analysis/AnalysisClient.js [hunk 2]
```diff
@@ -157,8 +199,15 @@ class AnalysisClient {
...
+
+    let categorizedItems = this._filterProductsToCategories(that.analyzedProductsInfo);
+    
+    //write all data into separate file for each category
+    this._writeAllCategories(categorizedItems);
+
+
     //write price analysis info for analyzed products
-    this._writeToFile("analyzed_items_info.txt");
+    //this._writeToFile("analyzed_items_info.txt");
```
Comment (danfstucky) [inline src/analysis/AnalysisClient.js]: Is there a reason you commented this out?

Comment (sulaimonlasisi) [reply src/analysis/AnalysisClient.js]: I forgot to delete it after testing.  Fixed here  9aec7ee

File: src/analysis/AnalysisClient.js [hunk 3]
```diff
@@ -157,8 +199,11 @@ class AnalysisClient {
         that.analyzedProductsInfo.push(analyzedProduct);
       } 
     });
-    //write price analysis info for analyzed products
-    this._writeToFile("analyzed_items_info.txt");
+
+    let categorizedItems = this._filterProductsToCategories(that.analyzedProductsInfo);
```
Comment (danfstucky) [inline src/analysis/AnalysisClient.js]: Seems odd that we would need to pass an instance variable to another function that also belongs to the class. Either it shouldn't be an instance variable, or it doesn't need to be passed to `_filterProductsToCategories` at all.

Comment (sulaimonlasisi) [reply src/analysis/AnalysisClient.js]: Fixed here: 8a911f8

**Review by danfstucky — APPROVED** [review]
+1

**Pull request merged**